### PR TITLE
A new way to get into the Hazardous Waste Sarcophagus 

### PR DESCRIPTION
**Pull request by thelonestander**
Adds a book that spawns with the military ID in the Hazardous Waste Sarcophagi.

SUMMARY: Content "Adds a new book that spawns with the military id card in the Hazardous Waste Sarcophagi overmap locations"

#### Purpose of change

1.) The main reason to change this is to give players the ability to go into the location and see what it may have downstairs. Im not spoiling anything, but its cool. 
2.) You can apparently access this location via a questline, however. I feel as though not everyone will find the quest giver (I don't even know who gives the quest) so I gave this as an extra option just in case players cant find it either. 
3.) Another good reason to have this as an option is to allow players to access what IS below and might be able to make use of it. Those of you who have been down to the basement know what loot is there. This may open up options, and make nanofab templets more...useable. 

#### Describe the solution

Added the new book as a dedicated spawn. In game youll ever only need one so this seems like a viable option. 
Also since it ONLY spawns in these locations this gives the people the *want* to loot them as well. 

#### Describe alternatives you've considered

None.

#### Testing

Tested in game, loaded up. Debug spawned the overmap special in and boom it spawned. Have fun guys! 

#### Additional context

Screenshot (808)

### CODE REVIEW

File: data/json/items/book/misc.json [hunk 1]
```diff
@@ -957,5 +957,24 @@
     "price": 4550,
     "looks_like": "tall_tales",
     "fun": 3
+  },
+  {
+    "id": "sa_cryptology_key",
+    "type": "BOOK",
+    "name": { "str_sp": "High Level Commercial and Military Cryptology" },
+    "description": "An incredibly dense volume describing different advanced cryptographical methods, their strenghts, and weaknesses. This text would probably be easier to digest with an added decade in Computer Science Academia.",
```
[the remark this answers is not in the document]
Comment (thelonestander) [reply data/json/items/book/misc.json:965]: Fixed! Forgot about this. Thank you.

File: data/json/items/book/misc.json [hunk 2]
```diff
@@ -957,5 +957,24 @@
     "price": 4550,
     "looks_like": "tall_tales",
     "fun": 3
+  },
+  {
+    "id": "sa_cryptology_key",
+    "type": "BOOK",
+    "name": { "str_sp": "High Level Commercial and Military Cryptology" },
+    "description": "An  incredibly  dense  volume  describing  different  advanced  cryptographical  methods,  their strenghts,  and weaknesses.  This  text  would  probably  be  easier  to  digest  with  an  added  decade  in  Computer  Science  Academia.",
```
Comment (Coolthulhu) [inline data/json/items/book/misc.json:965]: A whole load of extra spaces in there.

Comment (thelonestander) [reply data/json/items/book/misc.json]: Fixed and added spaces to "name" since its a descriptive string.

**Comment by olanti-p** [comment]
```
(json-error)
Json error: data/mods/dda/../../json/items/book/misc.json:965:135: insufficient spaces at this location.  2 required, but only 1 found.
    Suggested fix: insert " "
    At the following position (marked with caret)

    "type": "BOOK",
    "name": { "str_sp": "High  Level  Commercial  and  Military  Cryptology" },
    "description": "An  incredibly dense volume describing different advanced cryptographical methods, their strenghts, and weaknesses.
                                                                                                                                      ^
                                                                                                                                        This text would probably be easier to digest with an added decade in Computer Science Academia.",
    "weight": 3000,
    "volume": "2 L",
```

Bright Night - same as its ancestor, Dark Days Ahead, - uses monospaced fonts for its curses and SDL builds. Monospaced fonts are what typewriters used to use, and one of the conventions for text style typed with a typewriter was to use single space after a word and double space after a dot - for ease of reading. Same convention was adopted by DDA at some point in the past, and is enforced by automatic checks, and these checks were inherited by BN.

To satisfy the build checks, you'll have to use single space after words and double space after dots (after sentence ends, as @KheirFerrum pointed out).

**Comment by thelonestander** [comment]
> ```
> (json-error)
> Json error: data/mods/dda/../../json/items/book/misc.json:965:135: insufficient spaces at this location.  2 required, but only 1 found.
>     Suggested fix: insert " "
>     At the following position (marked with caret)
> 
>     "type": "BOOK",
>     "name": { "str_sp": "High  Level  Commercial  and  Military  Cryptology" },
>     "description": "An  incredibly dense volume describing different advanced cryptographical methods, their strenghts, and weaknesses.
>                                                                                                                                       ^
>                                                                                                                                         This text would probably be easier to digest with an added decade in Computer Science Academia.",
>     "weight": 3000,
>     "volume": "2 L",
> ```
> 
> Bright Nights - same as its ancestor, Dark Days Ahead, - uses monospaced fonts for its curses and SDL builds. Monospaced fonts are what typewriters used to use, and one of the conventions for text typed with a typewriter was to use single space after a word and double space after a dot - for ease of reading. Same convention was adopted by DDA at some point in the past, and is enforced by automatic checks, and these checks (and style) were inherited by BN.
> 
> To satisfy the build checks, you'll have to use single space after words and double space after dots (after sentence ends, as @KheirFerrum pointed out).

Oh dang it. Im so sorry. I read what he said wrong. It should be fixed now. Thank you Olanti

File: data/json/items/book/misc.json [hunk 3]
```diff
@@ -957,5 +957,24 @@
     "price": 4550,
     "looks_like": "tall_tales",
     "fun": 3
+  },
+  {
+    "id": "sa_cryptology_key",
+    "type": "BOOK",
+    "name": { "str_sp": "High Level Commercial and Military Cryptology" },
+    "description": "An  incredibly dense volume describing different advanced cryptographical methods, their strenghts, and weaknesses.  This text would probably be easier to digest with an added decade in Computer Science Academia.",
```
Comment (olanti-p) [inline data/json/items/book/misc.json:965]:
```suggestion
    "description": "An incredibly dense volume describing different advanced cryptographical methods, their strenghts, and weaknesses.  This text would probably be easier to digest with an added decade in Computer Science Academia.",
```
Leftover space

**Comment by Coolthulhu** [comment]
Wait, so this just adds a high skill computer book?

The description you provided makes it sound like something completely different.

**Comment by thelonestander** [comment]
> Wait, so this just adds a high skill computer book?
> 
> The description you provided makes it sound like something completely different.

Yeah, it adds a book to the Sarcophagi so the player can unlock the console and go down. In game you only need one of these codes to access all of them. Thats a vanilla thing. So I just added it to the location so players can go down without having to mine a bunch.

**Comment by Coolthulhu** [comment]
> In game you only need one of these codes to access all of them.

Oh, so the book is hardcoded to let you into the sarcophagus?

**Comment by thelonestander** [comment]
> > In game you only need one of these codes to access all of them.
> 
> Oh, so the book is hardcoded to let you into the sarcophagus?

I do believe so, the book doesn't just grant access, it gives you a recipe to craft the code to get it. So Youd still have to do a bit of work to get in. So, find book, craft code viola! Access to all the sarcophagi's!

**Comment by Coolthulhu** [comment]
> it gives you a recipe to craft the code to get it

Well, it doesn't do that bit.

**Comment by thelonestander** [comment]
> > it gives you a recipe to craft the code to get it
> 
> Well, it doesn't do that bit.

It should have. I fixed it, added the recipe. So you should now have to get the book, then craft the code.

**Pull request merged**